Xamarin.Forms.Core/Visuals/VisualTypeConverter.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Xamarin.Forms
+{
+	[Xaml.TypeConversion(typeof(IVisual))]
+	public class VisualTypeConverter : TypeConverter
+	{
+		static Dictionary<string, IVisual> _visualTypeMappings;
+		static Dictionary<string, IVisual> InitMappings()
+		{
+			var mappings = new Dictionary<string, IVisual>(StringComparer.OrdinalIgnoreCase);
+			_visualTypeMappings = mappings;
+			return mappings;
+		}
+
+		public static void Register(string visualKey, IVisual visual)
+		{
+			if (_visualTypeMappings == null)
+				InitMappings();
+
+			_visualTypeMappings[visualKey] = visual;
+		}
+
+		internal static void Register(Type visual)
+		{
+			if (_visualTypeMappings == null)
+				InitMappings();
+
+			if(!_visualTypeMappings.ContainsKey(visual.FullName))
+			{
+				try
+				{
+					IVisual registeredVisual = (IVisual)Activator.CreateInstance(visual);
+					_visualTypeMappings[visual.Name] = registeredVisual;
+					_visualTypeMappings[visual.FullName] = registeredVisual;
+					_visualTypeMappings[$"{visual.Name}Visual"] = registeredVisual;
+					_visualTypeMappings[$"{visual.FullName}Visual"] = registeredVisual;
```

Comment:
I added a few permutations of possible names people can use. If there are two Material visual libraries then people could use the full name of the type to resolve to what they want